lib/rules/no-lonely-if.js

```diff
@@ -52,9 +51,26 @@ module.exports = {
                                 return null;
                             }
 
+                            if (
+                                node.consequent.type !== "BlockStatement" && lastIfToken.value !== ";" && tokenAfterElseBlock &&
+                                (
+                                    node.consequent.loc.end.line === tokenAfterElseBlock.loc.start.line ||
+                                    /^[(\[+-`]/.test(tokenAfterElseBlock.value) ||
+                                    lastIfToken.value === "++" ||
+                                    lastIfToken.value === "--"
+                                )
+                            ) {
```

Comment:
I guess there are common conditions. Maybe...

```js
const ASI_DANGER_TOKENS = /^[([`+-]$/;
const DECLARATION_TYPES = /Declaration$/;

/**
 * Check whether it can remove the braces of the given `BlockStatement` node safely or not.
 *
 * This checks only inside of the block and around the closing brace.
 *
 * @param {ASTNode} node - The `BlockStatement` node to check.
 * @param {SourceCode} sourceCode - The source code object to get tokens.
 * @returns {boolean} `true` if it may be able to remove the braces.
 */
function canRemoveBlockMaybe(node, sourceCode) {
    assert(node.type === "BlockStatement");

    if (node.body.length !== 1 || DECLARATION_TYPES.test(node.body[0].type)) {
        return false;
    }

    const closingBrace = sourceCode.getLastToken(node);
    const last = sourceCode.getTokenBefore(closingBrace);
    const next = sourceCode.getTokenAfter(closingBrace);

    return (
        !next ||
        last.value === ";" ||
        (last.loc.end.line !== next.loc.start.line && !ASI_DANGER_TOKENS.test(next.value))
    );
}
```